ambry-network/src/main/java/com.github.ambry.network/BlockingChannelInfo.java

```diff
@@ -168,11 +168,11 @@ private BlockingChannel getBlockingChannelBasedOnPortType(String host, int port)
     if (this.port.getPortType() == PortType.PLAINTEXT) {
       channel = new BlockingChannel(host, port, config.connectionPoolReadBufferSizeBytes,
           config.connectionPoolWriteBufferSizeBytes, config.connectionPoolReadTimeoutMs,
-          config.connectionPoolConnectTimeoutMs);
+          config.connectionPoolConnectTimeoutMs, config.connectionPoolSocketTcpNoDelay);
     } else if (this.port.getPortType() == PortType.SSL) {
       channel = new SSLBlockingChannel(host, port, registry, config.connectionPoolReadBufferSizeBytes,
           config.connectionPoolWriteBufferSizeBytes, config.connectionPoolReadTimeoutMs,
-          config.connectionPoolConnectTimeoutMs, sslSocketFactory, sslConfig);
+          config.connectionPoolConnectTimeoutMs, config.connectionPoolSocketTcpNoDelay, sslSocketFactory, sslConfig);
```

Comment:
I wonder if it's worthwhile to pass the `config` into BlockingChannel and SSLBlockingChannel constructor. Thus, we can configure the parameters inside constructor and make function signature simpler.

Reply:
That would require more substantial changes. That was my first thought, and I even started doing that, and once I saw how many changes that involved across the board (especially in the tests) - I opted for passing a parameter instead.

Reply:
I've added a separate constructor, which takes ConnectionPoolConfig instance as a parameter and updated the BlockingChannelInfo accordingly.